django/core/management/commands/test.py

```diff
@@ -80,6 +81,18 @@ def handle(self, *test_labels, **options):
         from django.conf import settings
         from django.test.utils import get_runner
 
+        if has_unmigrated_models():
+            self.stdout.write(self.style.NOTICE(
+                "  Your models have changes that are not yet reflected "
+                "in a migration."
+            ))
+            self.stdout.write(self.style.NOTICE(
+                "  Run 'manage.py makemigrations' to make new "
+                "migrations, and then re-run 'manage.py test'"
+            ))
+
+            return
```

Comment:
I think a warning would be fine, but I don't see a need to completely prevent running the tests (just like you can use `runserver` if you have changes without migrations).